### PR TITLE
Fix nullIf function.

### PR DESCRIPTION
**Pull request by alexey-milovidov**
For changelog. Remove if this is non-significant change.

Category (leave one):
- Bug Fix

Short description (up to few sentences):
Fixed wrong behaviour of `nullIf` function for constant arguments.

Detailed description (optional):
For backports: just copy `nullIf.cpp` file as a whole.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
CC @YiuRULE

File: dbms/src/Functions/nullIf.cpp [hunk 1]
```diff
@@ -43,18 +43,17 @@ class FunctionNullIf : public IFunction
 
     void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count) override
     {
-        /// nullIf(col1, col2) == if(col1 != col2, col1, NULL)
+        /// nullIf(col1, col2) == if(col1 = col2, NULL, 1)
```
Comment (millin) [inline dbms/src/Functions/nullIf.cpp]: Maybe `nullIf(col1, col2) == if(col1 = col2, NULL, col1)` ?

Comment (alexey-milovidov) [reply dbms/src/Functions/nullIf.cpp]: :+1:

Comment (alexey-milovidov) [reply dbms/src/Functions/nullIf.cpp]: https://github.com/yandex/ClickHouse/commit/e573b06373c7215b25b5e2859575c1a3ccfeebb0

**Pull request merged**